lib/topological_inventory/ansible_tower/targeted_refresh/service_instance.rb

```diff
@@ -1,35 +1,43 @@
 require "topological_inventory/ansible_tower/collector"
 require "topological_inventory/providers/common/operations/source"
-require "topological_inventory/providers/common/operations/sources_api_client"
+require "topological_inventory/providers/common/mixins/sources_api"
+require "topological_inventory/providers/common/mixins/x_rh_headers"
 require "topological_inventory/ansible_tower/receptor/async_receiver"
 
 module TopologicalInventory
   module AnsibleTower
     module TargetedRefresh
       class ServiceInstance < TopologicalInventory::AnsibleTower::Collector
+        include TopologicalInventory::Providers::Common::Mixins::SourcesApi
+        include TopologicalInventory::Providers::Common::Mixins::XRhHeaders
+
         REFS_PER_REQUEST_LIMIT = 20
 
         def initialize(payload = {})
+          self.operation = 'ServiceInstance'
           self.params    = payload['params']
           self.source_id = payload['source_id']
 
+          self.identity = params.to_a.first.to_h['request_context']
           # TODO: add metrics exporter
           super(payload['source_uid'], nil)
         end
 
         # Entrypoint for 'ServiceInstance.refresh' operation
         def refresh
+          self.operation = "#{operation}#refresh"
+
           return if params_missing?
 
           # Connection settings
           set_connection_data!
-          logger.info("ServiceInstance#refresh - Connection set successfully")
+          logger.info_ext(operation, "Connection set successfully")
 
           # Get all tasks
           tasks, refresh_state_uuid = {}, SecureRandom.uuid
           params.to_a.each do |task|
             if task['task_id'].blank? || task['source_ref'].blank?
-              logger.warn("ServiceInstance#refresh - missing data for task: #{task}")
+              logger.warn_ext(operation, "Missing data for task: #{task}")
```

Comment:
I'm not seeing much gain between:
```
logger.warn_ext(operation, "Missing data for task: #{task}")
```
vs

```
logger.warn("ServiceInstance#refresh - missing data for task: #{task}")
```

The requirement to define `self.operation` on line 28 makes logging more complicated in my opinion.

Reply:
@syncrou You'll find the gain during troubleshooting in Kibana. By this, you're forced to specify the operation name, which is unique to all logs in operation so you will know that all operation-related logs will be filterable by one unique pattern

Reply:
I'm not going to hold up this PR. It's that these project specific logging wrappers need to be remembered by a human.

So I get why this is done, but I'm a big fan of how. I would rather we overloaded `logger.warn` and allow an accessor to pass in the object type. I even wouldn't mind a class called `KibanaLogger` which inherits from `Logger`

meta
```
logger = KibanaLogger.new(operation)
logger.warn("problem")
```

A separate method `warn_ext` get's difficult to remember to log in a specific way, and I'm worried we're going to miss it (when) those adding to this code base start to increase.

Reply:
yeah, the name isn't the best. I didn't realize this solution, I like it :+1: